apps/sasquatch/src/projectDependency/java/com/microsoft/appcenter/sasquatch/activities/StorageActivity.java

```diff
@@ -42,38 +52,138 @@
+
+        @Override
+        public void accept(PaginatedDocuments<TestDocument> documents) {
+            currentAppDocuments = documents;
+            updateAppDocument(documents.getCurrentPage().getItems());
+        }
+    };
+
+    private AppCenterConsumer<PaginatedDocuments<Map>> uploadUser = new AppCenterConsumer<PaginatedDocuments<Map>>() {
+
+        @Override
+        public void accept(PaginatedDocuments<Map> documents) {
+            currentUserDocuments = documents;
+            updateUserDocuments(documents.getCurrentPage().getItems());
+        }
+    };
+
+    private RecyclerView.OnScrollListener scrollAppListener = new RecyclerView.OnScrollListener() {
+
+        @Override
+        public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newState) {
+            super.onScrollStateChanged(recyclerView, newState);
+        }
+
+        @Override
+        public void onScrolled(@NonNull RecyclerView recyclerView, int dx, int dy) {
+            super.onScrolled(recyclerView, dx, dy);
+            if (currentAppDocuments != null && currentAppDocuments.hasNextPage() && !isLoading) {
+                isLoading = true;
+                currentAppDocuments.getNextPage().thenAccept(new AppCenterConsumer<Page<TestDocument>>() {
+                    @Override
+                    public void accept(Page<TestDocument> testDocumentPage) {
+                        isLoading = false;
+                        updateAppDocument(testDocumentPage.getItems());
+                    }
+                });
+            }
+        }
+    };
+
+    private RecyclerView.OnScrollListener scrollUserListener = new RecyclerView.OnScrollListener() {
+
+        @Override
+        public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newState) {
+            super.onScrollStateChanged(recyclerView, newState);
+        }
+
+        @Override
+        public void onScrolled(@NonNull RecyclerView recyclerView, int dx, int dy) {
+            super.onScrolled(recyclerView, dx, dy);
+            if (currentUserDocuments != null && currentUserDocuments.hasNextPage() && !isLoading) {
+                isLoading = true;
+                currentUserDocuments.getNextPage().thenAccept(new AppCenterConsumer<Page<Map>>() {
+
+                    @Override
+                    public void accept(Page<Map> mapPage) {
+                        updateUserDocuments(mapPage.getItems());
+                    }
+                });
+            }
+        }
+    };
+
+    private void updateUserDocuments(List<Document<Map>> documents) {
+        if (documents == null)
+            return;
+
+        for (Document<Map> document : documents) {
+            sUserDocumentList.add(String.format("%s_%s", document.isFromCache() ? CACHED_PREFIX : REMOTE_PREFIX, document.getId()));
+            mUserDocumentContents.add(Utils.getGson().toJson(document.getDocument()));
+        }
+        saveArrayToPreferences(sUserDocumentList, USER_DOCUMENT_LIST);
```

Comment:
Don't cache lists

Reply:
:+1: already discussed this with storage team.